stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java

```diff
@@ -94,14 +112,96 @@ public static MimeMessage generateMessage(Session session, String username, MapV
             emailMessage.setReplyTo(replyToAddressArray);
         }
         emailMessage.setSubject(subject);
-        emailMessage.setText(messageBody);
         emailMessage.setFrom(new InternetAddress(fromAddress));
         if (!senderAddress.isEmpty()) {
             emailMessage.setSender(new InternetAddress(senderAddress));
         }
+        ArrayValue attachments = message.getArrayValue(EmailConstants.MESSAGE_ATTACHMENTS);
+        if (attachments == null) {
+            emailMessage.setText(messageBody);
+        } else {
+            addBodyAndAttachments(emailMessage, messageBody, attachments);
+        }
         return emailMessage;
     }
 
+    private static void addBodyAndAttachments(MimeMessage emailMessage, String messageBody, ArrayValue attachments)
+            throws MessagingException, IOException {
+        BodyPart messageBodyPart = new MimeBodyPart();
+        messageBodyPart.setText(messageBody);
+        Multipart multipart = new MimeMultipart();
+        multipart.addBodyPart(messageBodyPart);
+        for (int i = 0; i < attachments.size(); i++) {
+            if (attachments.get(i) instanceof ObjectValue) {
+                ObjectValue mimeEntity = (ObjectValue) attachments.get(i);
+                String contentType = getContentTypeWithParameters(mimeEntity);
+                if (contentType.startsWith(MimeConstants.MULTIPART_AS_PRIMARY_TYPE)) {
+                    multipart.addBodyPart(populateMultipart(mimeEntity));
+                } else {
+                    multipart.addBodyPart(buildJavaMailBodyPart(mimeEntity, contentType));
+                }
+            }
+        }
+        emailMessage.setContent(multipart);
+    }
+
+    private static MimeBodyPart populateMultipart(ObjectValue mimeEntity) throws IOException, MessagingException {
+        Multipart multipart = new MimeMultipart();
+        ArrayValue multipartMimeEntityArrayValue = EntityBodyHandler.getBodyPartArray(mimeEntity);
+        int entityCount = multipartMimeEntityArrayValue.size();
+        for (int i = 0; i < entityCount; i++) {
```

Comment:
Can be replaced with a `for each` loop ?

Reply:
`for each` loop is not supporting `ArrayValue` type by default. Therefore, it is better to use a counter to access the `ObjectValue` entries with the `for` loop.